ledger.rb

```diff
@@ -1,24 +1,156 @@
+  class_option :file, type: :string, default: 'index.ledger'
+
+  def initialize(*args)
+    super
+    @parser = Parser.new
+    @parser.parse_ledger(options[:file])
+    @balances = {}
+  end
+
   desc "register", "The register command displays all the postings occurring in a single account, line by line."
   def register
-    puts "register command"
+    @parser.parsed_file.each do |transaction|
+      register_title_line(transaction[:date], transaction[:description])
+
+      transaction[:accounts].each do |account|
+        if @balances.include?(account[:currency])
+          @balances[account[:currency]] += account[:amount]
+        else
+          @balances[account[:currency]] = account[:amount]
+        end
+
+        @balances[account[:currency]] = @balances[account[:currency]].round(2)
+        register_line(account[:description], account[:amount], @balances, account[:currency])
+      end
+    end
+
+    register_balance(@balances)
   end
 
   desc "balance", "The balance command reports the current balance of all accounts. "
   def balance
-    parser = Parser.new
-    parsed_info = parser.parse_ledger("index.ledger")
+    transactions = {}
+    transactions_sums = {}
+
+    @parser.parsed_file.each do |transaction|
+      transaction[:accounts].each do |account|
+        if @balances.include?(account[:currency])
+          @balances[account[:currency]] += account[:amount]
+        else
+          @balances[account[:currency]] = account[:amount]
+        end
+
+        @balances[account[:currency]] = @balances[account[:currency]].round(2)
+
+        if transactions_sums.include?(account[:description])
+          transactions_sums[account[:description]] += account[:amount]
+          transactions[account[:description]] = full_action(transactions_sums[account[:description]], account[:currency])
+        else
+          transactions_sums[account[:description]] = account[:amount]
+          transactions[account[:description]] = full_action(account[:amount], account[:currency])
+        end
+      end
+    end
+
+    transactions.sort_by { |k, v| v }.reverse.each do |description, action|
+      balance_line(action, description)
+    end
+
+    puts '--------------------'
+
+    @balances.each do |k, v|
+      balance_line(full_action(v, k))
+    end
 
-    puts parsed_info
   end
 
   desc "print", "The print command prints out ledger transactions in a textual format that can be parsed by Ledger."
   def print
-    puts "print command"
+    @parser.parsed_file.each do |transaction|
+      print_title_line(transaction[:date], transaction[:description])
+
+      transaction[:accounts].each do |account|
+        print_line(account[:description], account[:amount], account[:currency])
+      end
+
+      puts "\n"
+    end
+
+  end
+
+  private
+
+  def full_action(amount, currency)
+    "#{'%.2f' % amount} #{currency}"
+  end
+
+  def balance_line(action, description = nil)
+    action = " " * (20 - action.size) + action + ' '
```

Comment:
let's not update the parameters you are sending to a method, you can create a new local variable here